Add query caching to zGetAttachmentIdByUrl function

alessandrocarrera opened the pull request:
In this PR I added query caching for improving the performance of `zGetAttachmentIdByUrl` function.

alessandrocarrera committed:
perf(zgetattachmentidbyurl): add query caching

In this commit I added query caching for improving the performance of zGetAttachmentIdByUrl
function.

HyprCoder commented on categories-images.php:
```diff
@@ -198,12 +198,34 @@ function zSaveTaxonomyImage($term_id) {
-        return (!empty($id)) ? $id : NULL;
+
+        // Convert and sanitize image url into md5 string.
+        $md5_image_src = md5( esc_url_raw( $image_src ) );
```
proviamo con sha256, come consigliato da Codacy

alessandrocarrera replied:
Done! Is it a problem the sha256 for the cache key name?

alessandrocarrera replied:
@HyprCoder I think that is not a problem because the maximum key size for Redis is 512MB:

https://redis.io/topics/data-types-intro#redis-keys

What do you think?

alessandrocarrera committed:
refactor(zgetattachmentidbyurl): change coding algorithm

In this commit I changed md5 with sha256 for the key cache name and the name of the variable $id to
$id_img for having 3 and more char in the name.
